tfchain_bridge/pkg/bridge/bridge.go

```diff
@@ -101,467 +87,80 @@ func (bridge *Bridge) Start(ctx context.Context) error {
 	if err != nil {
 		return errors.Wrap(err, "failed to get block height from persistency")
 	}
-
-	cl, _, err := bridge.subClient.GetClient()
+	log.Info().Msg("starting stellar subscription...")
+	stellarSub, err := bridge.wallet.StreamBridgeStellarTransactions(ctx, height.StellarCursor)
 	if err != nil {
-		return errors.Wrap(err, "failed to get client")
+		return errors.Wrap(err, "failed to monitor bridge account")
 	}
 
-	chainHeadsSub, err := cl.RPC.Chain.SubscribeFinalizedHeads()
+	log.Info().Msg("starting tfchain subscription...")
+	tfchainSub, err := bridge.subClient.SubscribeTfchainBridgeEvents(ctx)
 	if err != nil {
-		return errors.Wrap(err, "failed to subscribe to finalized heads")
+		return errors.Wrap(err, "failed to subscribe to tfchain")
 	}
 
-	go func() {
-		log.Info().Msg("starting minting subscription...")
-		if err := bridge.wallet.MonitorBridgeAccountAndMint(ctx, bridge.mint, height.StellarCursor); err != nil {
-			panic(err)
-		}
-	}()
-
 	for {
 		select {
-		case head := <-chainHeadsSub.Chan():
-			err := bridge.processEventsForHeight(uint32(head.Number))
-			if err != nil {
-				return err
+		case data := <-tfchainSub:
+			if data.Err != nil {
+				return errors.Wrap(err, "failed to process events")
+			}
+			for _, withdrawCreatedEvent := range data.Events.WithdrawCreatedEvents {
+				err := bridge.handleWithdrawCreated(ctx, withdrawCreatedEvent)
+				if err != nil {
+					return errors.Wrap(err, "failed to handle withdraw created")
+				}
+			}
+			for _, withdrawExpiredEvent := range data.Events.WithdrawExpiredEvents {
+				err := bridge.handleWithdrawExpired(ctx, withdrawExpiredEvent)
+				if err != nil {
+					return errors.Wrap(err, "failed to handle withdraw expired")
+				}
+			}
+			for _, withdawReadyEvent := range data.Events.WithdrawReadyEvents {
+				err := bridge.handleWithdrawReady(ctx, withdawReadyEvent)
+				if err != nil {
+					return errors.Wrap(err, "failed to handle withdraw ready")
+				}
+			}
+			for _, refundReadyEvent := range data.Events.RefundReadyEvents {
+				err := bridge.handleRefundReady(ctx, refundReadyEvent)
+				if err != nil {
+					return errors.Wrap(err, "failed to handle refund ready")
+				}
+			}
+			for _, refundExpiredEvent := range data.Events.RefundExpiredEvents {
+				err := bridge.handleRefundExpired(ctx, refundExpiredEvent)
+				if err != nil {
+					return errors.Wrap(err, "failed to handle refund expired")
+				}
+			}
+		case data := <-stellarSub:
+			if data.Err != nil {
+				return errors.Wrap(err, "failed to get mint events")
+			}
+
+			for _, mEvent := range data.Events {
+				err := bridge.mint(mEvent.Senders, mEvent.Tx)
+				for err != nil {
+					log.Err(err).Msg("Error occured while minting")
+					if errors.Is(err, pkg.ErrTransactionAlreadyRefunded) {
```

Comment:
Now it just logs "error occured while minting", would you log the fact that the refund already happened?